File: fixtures/javascript/controller-service-provider/src/service/paymentService.js
// Service layer wired with CommonJS require bindings.

const { StripeProvider } = require("../provider/stripeProvider.js");
const { sharedUniqueTarget } = require("../shared.js");

class PaymentService {
    constructor() {
        this.provider = new StripeProvider("test-key");
    }

    refund(amountCents) {
        sharedUniqueTarget();
        this.provider.refund(amountCents);
    }
}

function buildPaymentService() {
    return new PaymentService();
}

module.exports = { PaymentService, buildPaymentService };

File: fixtures/python/controller-service-provider/src/service/payment_service.py
from provider.provider import PaymentProvider
from provider.stripe_provider import StripeProvider
from shared import shared_unique_target


class PaymentService:
    def __init__(self):
        self.provider = StripeProvider("test-key")

    def refund(self, amount_cents):
        shared_unique_target()
        self.provider.refund(amount_cents)


def build_payment_service():
    return PaymentService()
